python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -675,15 +687,26 @@ cdef cudf_io_types.compression_type _get_comp_type(object compression):
         raise ValueError("Unsupported `compression` type")
 
 
-cdef _set_col_metadata(Column col, column_in_metadata& col_meta):
+cdef _set_col_metadata(
+    Column col,
+    column_in_metadata& col_meta,
+    object nullability
+):
+    if col.nullable and nullability is False:
+        warnings.warn(f"column {col_meta.get_name().decode()} contains null "
```

Comment:
How do we decide if this is worth issuing a warning? It seems like there are valid use cases for nullability=False that are intentional and not deserving of a warning.

Reply:
> It seems like there are valid use cases for nullability=False

I'm not sure about this. Columns without nulls would have been written as non-nullable in Parquet(?) even without `nullability=False`, and the option will be ignored for columns with nulls. So unless we are passing this to the chunked writer, I don't think `nullability=False` does anything.

Reply:
Let's align with @vuule on the C++ side -- I am fine with these names but would prefer an `enum class` instead of `bool` on the C++ side for alignment (if it's currently a `bool` -- can't recall).

Reply:
It's currently a bool on c++ side.

Reply:
That sounds like an improvement to me. Let's do that. `force_nullable=True` or `force_nullable=False`, default is `False`.

Reply:
I went ahead with `force_nullable_schema` option and updated the docs.